v3/src/data-interactive/handlers/collection-handler.ts

```diff
@@ -23,6 +23,14 @@ export const diCollectionHandler: DIHandler = {
 
     const newCollections: ICollectionModel[] = []
     dataContext.applyModelChange(() => {
+      // Remove the default collection if it exists and is empty
+      // This is a special case and it might make more sense to do this in a more generic way,
+      // like a reaction that removes empty collections whenever attributes change.
+      if (dataContext.collections.length === 1) {
+        const oldCollection = dataContext.collections[0]
+        if (oldCollection.attributes.length === 0) dataContext.removeCollection(oldCollection)
+      }
```

Comment:
It is possible that the loop below won't create any collections (e.g. if none of them have a `name`), in which case the `DataSet` would be left with no collections. You can rectify this by making it a local function and then calling it just before `dataContext.addCollection()` in line 69 below, so that empty collections are deleted only when adding new collections. I'm intrigued by the idea of automating this, but since users can't create collections on their own, I think handling this special case here seems acceptable.